app/Http/Middleware/CustomApiTokenAuth.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+use Illuminate\Http\Request;
+use App\Models\User;
+use Illuminate\Auth\AuthenticationException;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Log;
+
+class CustomApiTokenAuth
+{
+    /**
+     * Handle an incoming request.
+     * This middleware will try multiple methods of authentication to ensure API requests succeed.
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        // STEP 1: Log detailed request information
+        Log::debug("API Request", [
```

Comment:
Is logging this some sort of security concern?

Reply:
This would only show up if we set `LOG_LEVEL=debug` and even then it would get logged to a log file.